skyportal/handlers/api/gcn.py

```diff
@@ -1033,9 +1056,98 @@ def add_tiles(localization_id):
+                                DefaultObservationPlanRequest.default_survey_efficiencies
+                            )
+                        ],
+                    )
+                )
+            )
+            .unique()
+            .all()
+        )
+        gcn_observation_plans = []
+        for plan in default_observation_plans:
+            allocation = session.scalars(
+                Allocation.select(user).where(Allocation.id == plan.allocation_id)
+            ).first()
+
+            gcn_observation_plan = {}
+            gcn_observation_plan['allocation_id'] = allocation.id
+            gcn_observation_plan['payload'] = plan.payload
+            gcn_observation_plan['survey_efficiencies'] = [
+                survey_efficiency.to_dict()
+                for survey_efficiency in plan.default_survey_efficiencies
+            ]
+            gcn_observation_plans.append(gcn_observation_plan)
+        gcn_observation_plans = gcn_observation_plans + config_gcn_observation_plans
+
+        event = session.scalars(
+            GcnEvent.select(user).where(GcnEvent.dateobs == localization.dateobs)
+        ).first()
+        start_date = str(datetime.datetime.utcnow()).replace("T", "")
+        end_date = str(datetime.datetime.utcnow() + datetime.timedelta(days=1)).replace(
+            "T", ""
+        )
+
+        for ii, gcn_observation_plan in enumerate(gcn_observation_plans):
+            allocation_id = gcn_observation_plan['allocation_id']
+            allocation = session.scalars(
+                Allocation.select(user).where(Allocation.id == allocation_id)
+            ).first()
+
+            if allocation is not None:
+                payload = {
+                    **gcn_observation_plan['payload'],
+                    'start_date': start_date,
+                    'end_date': end_date,
+                    'queue_name': f'{allocation.instrument.name}-{start_date}-{ii}',
+                }
+                plan = {
+                    'payload': payload,
+                    'allocation_id': allocation.id,
+                    'gcnevent_id': event.id,
+                    'localization_id': localization.id,
+                }
+
+                plan_id = post_observation_plan(
+                    plan, user_id, session, asynchronous=False
+                )
+                for survey_efficiency in gcn_observation_plan['survey_efficiencies']:
+                    post_survey_efficiency_analysis(
+                        survey_efficiency, plan_id, user_id, session, asynchronous=False
+                    )
+
+        return log(
+            f"Generated tiles / observation plans for localization {localization_id}"
+        )
     except Exception as e:
-        log(f"Unable to generate contour for localization {localization_id}: {e}")
+        log(
+            f"Unable to generate tiles / observation plans for localization {localization_id}: {e}"
+        )
     finally:
         Session.remove()
```

Comment:
Any reason why in the `add_tiles_and_observation_plans(localization_id, user_id)` method you choose to create a session and remove it manually rather than using a context manager as we usually do?

Reply:
We have done this for all our async handlers... not sure about why honestly :)